src/PrestaShopBundle/Controller/Admin/Sell/Catalog/AttributeGroupController.php

```diff
@@ -96,12 +123,40 @@ public function createAction()
      *
      * @param int $attributeGroupId
      *
-     * @return RedirectResponse
+     * @return Response
      */
-    public function editAction(int $attributeGroupId)
+    public function editAction(Request $request, int $attributeGroupId): Response
     {
-        //@todo: implement in antoher pr
-        return $this->redirectToRoute('admin_attribute_groups_index');
+        $attributeGroupFormBuilder = $this->get('PrestaShop\PrestaShop\Core\Form\IdentifiableObject\Builder\AttributeGroupFormBuilder');
+        $attributeFormHandler = $this->get('PrestaShop\PrestaShop\Core\Form\IdentifiableObject\Handler\AttributeGroupFormHandler');
+
+        $attributeGroupForm = $attributeGroupFormBuilder->getFormFor($attributeGroupId);
+        $attributeGroupForm->handleRequest($request);
+
+        try {
+            $handlerResult = $attributeFormHandler->handleFor($attributeGroupId, $attributeGroupForm);
+
+            if (null !== $handlerResult->getIdentifiableObjectId()) {
+                $this->addFlash('success', $this->trans('Successful update', 'Admin.Notifications.Success'));
+
+                return $this->redirectToRoute('admin_attribute_groups_index');
+            }
+        } catch (Exception $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+        }
+
+        return $this->render(
+            '@PrestaShop/Admin/Sell/Catalog/AttributeGroup/edit.html.twig',
+            [
+                'layoutTitle' => $this->trans(
+                    'Editing attribute %name%',
+                    'Admin.Navigation.Menu',
+                    ['%name%' => $attributeGroupForm->getData()['name'][$this->getContextLangId()]]
```

Comment:
Maybe the name could be "extracted" before in a variable to make the code a bit cleaner, but mostly we could anticipate in case the localized array doesn't contain a value for the context's language, we can use the first one instead:
```php
$formData = $attributeGroupForm->getData();
$attributeGroupName = $formData['name'][$this->getContextLangId()]] ?? reset($formData['name']);
```

Reply:
Nice idea regarding context language!